File: app/src/test/java/us/hnry/fancy/utils/QuoteQueryBuilderTest.java
package us.hnry.fancy.utils;

import org.junit.Test;

import static junit.framework.Assert.assertEquals;

/**
 * @author Henry
 *         10/7/2017
 */
public class QuoteQueryBuilderTest {
    private String baseQuery = "select * from yahoo.finance.quotes where symbol in ";

    @Test
    public void testQuoteQueryBuilder() {
        assertEquals(baseQuery + "(\"APPL\")", new QuoteQueryBuilder("APPL").build());
    }

    @Test
    public void testQuoteQueryBuilderUpperCaseRule() {
        assertEquals(baseQuery + "(\"APPL\")", new QuoteQueryBuilder("appl").build());
    }

    @Test
    public void testQuoteQueryBuilder_sortMultipleSymbols() {
        assertEquals(baseQuery + "(\"APPL\",\"SQ\")", new QuoteQueryBuilder("sq", "appl").build());
    }

    @Test
    public void testQuoteQueryBuilderKt() {
        assertEquals(baseQuery + "(\"APPL\")", new QuoteQueryBuilder("APPL").build());
    }

    @Test
    public void testQuoteQueryBuilder_noArgs() {
        assertEquals(baseQuery + "()", new QuoteQueryBuilder().build());
    }
}

File: app/src/test/java/us/hnry/fancy/utils/QuoteQueryBuilderTest.kt
package us.hnry.fancy.utils

import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * @author Henry
 * *         10/7/2017
 */
class QuoteQueryBuilderTest {
    private val baseQuery = "select * from yahoo.finance.quotes where symbol in "

    @Test
    fun testQuoteQueryBuilder() {
        assertEquals("$baseQuery(\"APPL\")", QuoteQueryBuilder("APPL").build())
    }

    @Test
    fun testQuoteQueryBuilderUpperCaseRule() {
        assertEquals("$baseQuery(\"APPL\")", QuoteQueryBuilder("appl").build())
    }

    @Test
    fun testQuoteQueryBuilder_sortMultipleSymbols() {
        assertEquals("$baseQuery(\"APPL\",\"SQ\")", QuoteQueryBuilder("sq", "appl").build())
    }

    @Test
    fun testQuoteQueryBuilderKt() {
        assertEquals("$baseQuery(\"APPL\")", QuoteQueryBuilder("APPL").build())
    }

    @Test
    fun testQuoteQueryBuilder_noArgs() {
        assertEquals(baseQuery + "()", QuoteQueryBuilder().build())
    }
}